exercises/diffie-hellman/diffie-hellman.spec.js

```diff
@@ -0,0 +1,67 @@
+import DiffieHellman from './diffie-hellman';
+
+describe('diffie-hellman', () => {
+  const p = 23;
+  const g = 5;
+  let diffieHellman = new DiffieHellman(p, g);
+
+  const alicePrivateKey = 6;
+  const alicePublicKey = 8;
+
+  const bobPrivateKey = 15;
+  const bobPublicKey = 19;
+
+  test('throws an error if the constructor arguments are out of range', () => {
+
+    expect(() => {
+      new DiffieHellman(0, 9999)
+    }).toThrow();
+
+  });
+
+  test('throws an error if the constructor arguments are not prime', () => {
+
+    expect(() => {
+      new DiffieHellman(10, 13)
+    }).toThrow();
+
+  });
+
+  test('throws an error if private key >= p-1', () => {
```

Comment:
I usually favor one test for one assert, or just one assert by test. But it's not a golden rule. In this case, it makes sense to check `-1`, `0` and `1` in the same test (and `p` and `p + 1` in another one).

I commented about more than one case, not just `a < 2` because negative numbers, `0`  and `1` have *special* meanings. It's not that you have to test all possible values: `-100`, `-99`,... But `0`, `1` and so on are kind of edge cases, or very common edge cases.